src/extensions/query-translator/QueryTranslatorConfig.ts

```diff
@@ -0,0 +1,111 @@
+import { SELECTION_TYPES } from '../../config/CardConfig';
+import { ModelClient } from './clients/ModelClient';
+import { OpenAiClient } from './clients/OpenAiClient';
+import { VertexAiClient } from './clients/VertexAiClient';
+
+interface ClientSettingEntry {
+  label: string;
+  type: SELECTION_TYPES;
+  default: any;
+  authentication?: boolean; // Required for authentication, the user should insert all the required fields before trying to authenticate
+  hasAuthButton?: boolean; // Append a button at the end of the selector to trigger an auth request.
+  methodFromClient?: string; // String that contains the name of the client function to call to retrieve the data needed to fill the option
+}
+
+interface ClientSettings {
+  apiKey: ClientSettingEntry;
+  modelType: ClientSettingEntry;
+  region?: ClientSettingEntry;
+}
+
+interface ClientConfig {
+  clientName: string;
+  clientClass: ModelClient;
+  clientSettingsModal: JSX.Element;
+  settings: ClientSettings;
+}
+
+interface AvailableClients {
+  OpenAI: ClientConfig;
+  vertexAi: ClientConfig;
+}
+
+interface QueryTranslatorConfig {
+  availableClients: AvailableClients;
+}
+
+export const QUERY_TRANSLATOR_CONFIG: QueryTranslatorConfig = {
+  availableClients: {
+    OpenAI: {
+      clientName: 'OpenAI',
+      clientClass: OpenAiClient,
```

Comment:
Maybe be consistent and rename the class to OpenAIClient?

Reply:
The name that you see there is the now that will be shown to the final user, so I prefer sticking to just OpenAi